app/controllers/works_controller.rb

```diff
@@ -0,0 +1,90 @@
+class WorksController < ApplicationController
+  def index
+    @books = Work.sort_by_category("book")
+    @movies = Work.sort_by_category("movie")
+    @albums = Work.sort_by_category("album")
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      head :not_found
```

Comment:
In the future I advise just redirecting the user rather than just presenting them with a blank white page.